src/engine/ReactPlugin/ToolReactPlugin.js

```diff
@@ -24,18 +24,19 @@ export function withToolPlugin(ReactComponent, methods = []) {
   const RefComponent = forwardRef((props, ref) => ReactComponent(props, ref));
 
   const ToolComponent = forwardRef((props, ref) => {
-    const { on, off } = props;
+    const { on, off, profile } = props;
 
     /**
      * Component did mount
      */
     useEffect(() => {
       PluginManagerIDE.resetBookmarks();
-      on(PLUGINS.TABS.NAME, PLUGINS.TABS.ON.ACTIVE_TAB_CHANGE, async () => {
-        PluginManagerIDE.resetBookmarks();
+      on(PLUGINS.TABS.NAME, PLUGINS.TABS.ON.ACTIVE_TAB_CHANGE, async ({id}) => {
+        if(profile.name === id) PluginManagerIDE.resetBookmarks();
       });
 
       return () => {
+        console.log(">>> debug toolplugin unsubscribe ACTIVE_TAB_CHANGE")
         off(PLUGINS.TABS.NAME, PLUGINS.TABS.ON.ACTIVE_TAB_CHANGE);
       };
     }, []);
```

Comment:
maybe add profile.name to dependency array

Reply:
resolved